<?xml version="1.0" ?>
<robot xmlns:xacro="http://wiki.ros.org/xacro">

    <ros2_control name="GazeboSystem" type="system">
        <hardware>
            <plugin>gazebo_ros2_control/GazeboSystem</plugin>
        </hardware>

        <!-- ARM JOINTS -->
        <joint name="link1_to_link2">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <joint name="link2_to_link3">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <joint name="link3_to_link4">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <joint name="link4_to_link5">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <joint name="link5_to_link6">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <joint name="link6_to_flange_link">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <!-- GRIPPER MAIN JOINT -->
        <joint name="gripper_controller">
            <command_interface name="position"/>
            <state_interface name="position"/>
            <state_interface name="velocity"/>
        </joint>

        <!-- DO NOT ADD MIMIC JOINTS HERE -->
        <joint name="gripper_base_to_gripper_left2">
                <param name="mimic">gripper_controller</param>
                <param name="multiplier">1.0</param>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="gripper_left3_to_gripper_left1">
                <param name="mimic">gripper_controller</param>
                <param name="multiplier">-1.0</param>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="gripper_base_to_gripper_right3">
                <param name="mimic">gripper_controller</param>
                <param name="multiplier">-1.0</param>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="gripper_base_to_gripper_right2">
                <param name="mimic">gripper_controller</param>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>

            <joint name="gripper_right3_to_gripper_right1">
                <param name="mimic">gripper_controller</param>
                <param name="multiplier">1.0</param>
                <state_interface name="position"/>
                <state_interface name="velocity"/>
            </joint>
    </ros2_control>


    <gazebo>
        <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
            <parameters>$(find robot_description)/config/controllers.yaml</parameters> 
        <initial_joint_positions>
            <joint name="link1_to_link2" value="1.57"/>
            <joint name="link2_to_link3" value="-1.0"/>
            <joint name="link3_to_link4" value="0.5"/>
            <joint name="link4_to_link5" value="1.0"/>
            <joint name="link5_to_link6" value="0.0"/>
            <joint name="link6_to_flange_link" value="0.0"/>
            <joint name="gripper_controller" value="0.0"/>
            
        </initial_joint_positions>
 
        </plugin>
    </gazebo>

</robot>